topdown/glob.go

```diff
@@ -64,6 +66,13 @@ func globCompileAndMatch(id, pattern, match string, delimiters []rune) (bool, er
 		if p, err = glob.Compile(pattern, delimiters...); err != nil {
 			return false, err
 		}
+		if len(globCache) >= globCacheMaxSize {
+			// Delete a (semi-)random key to make room for the new one.
```

Comment:
why semi? I think it's pretty random 😅

Reply:
It's not explicitly random, as the randomness is an implementation detail of the map. 
I also think that it'd be deterministic over multiple runs with the same map entries.

Reply:
So we're checking `len(globCache) >= globCacheMaxSize` but we only remove one. This is the only code path where cached patterns are inserted, right? So it cannot happen that through some other code path, the cache would still continue to grow without bounds because we're only removing _one element_ here.

Reply:
Correct, discounting tests, this is the only place we touch the cache.